src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -585,7 +585,13 @@ class PartitionClauseSegment(BaseSegment):
         "PARTITION",
         "BY",
         Indent,
-        Delimited(Ref("ExpressionSegment"), delimiter=Ref("CommaSegment")),
+        OneOf(
+            # Brackets are optional in a partition by statement
+            Bracketed(
```

Comment:
This should be a pattern which exists elsewhere, but I couldn't find it, so just went with a OneOf with one bracketed and one not bracketed. We could make an 'OptionallyBracketed' segment for this purpose to be used in group by, partition by and other clauses which I assume are also currently unparsable with brackets. Have I missed something here @alanmcruickshank ?